tests/ci/run_pytest.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+run_pytest.py is the script submitted to Azure ML that runs pytest.
+pytest runs all tests in the specified test folder unless parameters
+are set otherwise.
+"""
+
+import os
+import argparse
+import subprocess
+
+from azureml.core import Run
+
+
+def create_arg_parser():
+    parser = argparse.ArgumentParser(description='Process inputs')
+    # test folder
+    parser.add_argument("--testfolder", "-f",
+                        action="store",
+                        default="./tests/unit",
+                        help="Folder where tests are located")
+    # internal test purposes
+    parser.add_argument("--num",
+                        action="store",
+                        default="99",
+                        help="test num")
+    # test markers
+    parser.add_argument("--testmarkers", "-m",
+                        action="store",
+                        default="not notebooks and not spark and not gpu",
```

Comment:
this will fail unless the user input is enclosed in quotes, another way might be to have flags for each option

```
parser.add_argument("--notebooks", "-n", action="store_true")
parser.add_argument("--spark", "-s", action="store_true")
parser.add_argument("--gpu", "-g", action="store_true")
...
test_marker = ' and '.join([name if getattr(args, name) else 'not ' + name for name in ['notebooks', 'spark', 'gpu'])
```

Reply:
There's a difference as the args are passed to AML as part of ScriptRunConfig.

To demonstrate this works,  testmarkers are deleted from the devops pipeline where python is called and it then uses the parser.add_argument:
 https://dev.azure.com/best-practices/recommenders/_build/results?buildId=1997&view=ms.vss-test-web.build-test-results-tab